### DISCUSSION TITLE
Image management && API access

### DISCUSSION THREAD

**Discussion by HenAmar** [ubicloud/ubicloud#1184] [Q&A]
I was exploring the usage of a hosted cloud with OVH baremetal machines. I followed the 'Build Your Own Cloud' guide. Couple of questions:

1. I've noticed that, as of now, the only available boot images are Ubuntu Jammy 22.04 LTS and AlmaLinux 9.1. I'm curious to know if there are any options for image management or if there are future plans regarding this.
2. I haven't come across any documented API for creating/destroying VMs. Does such an API exist?
3. I tried to "cloudify" ovh machine. However, the only option available when running the cloudify script was Hetzner. Is that the only vendor supported ?
4. I've continued and just mark  Hetzner Findland, But I still can't see the instance. Where on control pane it should appear ?

**Comment by furkansahin** [comment]
Hello @HenAmar ,
Thanks for trying out Ubicloud! 
> I've noticed that, as of now, the only available boot images are Ubuntu Jammy 22.04 LTS and AlmaLinux 9.1. I'm curious to know if there are any options for image management or if there are future plans regarding this.

This is a feature we are keeping an eye on and possibly will start implementing in near future. However, it is not planned for this quarter at the moment. What is your main usecase here? Would you like to have custom images built or more OS image choices? 

> I haven't come across any documented API for creating/destroying VMs. Does such an API exist?

We are working on our APIs at the moment. Soon, we will have the preview version. In the meantime, you can use the ubicloud console to create/destroy VMs.

> I tried to "cloudify" ovh machine. However, the only option available when running the cloudify script was Hetzner. Is that the only vendor supported?

Currently, our main focus is on the hosted version that you can find here [Ubicloud Console](console.ubicloud.com). That one is working on Hetzner and we are looking for including more providers. The Hetzner choice on the script is a mere placeholder which you can just pick for any server, however, you may encounter operational issues since it is a demo version. In full transparency, the self-hosted version isn't an immediate area of focus yet. We're looking to improve Ubicloud on the managed service side first; and then carry that over to the self-hosted version. Managed service side is fully operational on Hetzner.

> I've continued and just mark Hetzner Findland, But I still can't see the instance. Where on control pane it should appear ?

The hosts will not appear in the console, only the VMs. You can see the hosts in the `pry` console after cloudifying it. 

Hope I could answer all your questions. Please let us know if you need further information or want to give a feedback :)

Best,
Furkan

**Reply by HenAmar** [reply]
Thanks @furkansahin for the answers. 

In respect to the use case, I'm just digging to see if this could be a real solution which might facilitate the scheduling/managing of my heavy loaded CI jobs. Some CI jobs has to run directly on the VM (not containers) and I would like to chose the infra. 

In respect to the appearance of the hosts, since the Host is not available in the console, I wonder if there is any way I can choose on which host the VM will be created. Does capability exist ?

**Reply by fdr** [reply]
Currently, there is not such a feature, the allocator finds capacity (that is: it runs a SQL query) and puts the VM where-ever it is available.
